### ISSUE TITLE
Enums and custom display templates are broken in 1.1

### ISSUE THREAD

**Issue by tverboon** [aspnet/Mvc#5547]
We have some custom display templates for enums and display them like this:

View.cshtml:
`@Html.DisplayFor(x => x.DayOfWeek)`

DayOfWeek.cshtml in ~/Shared/DisplayTemplates/:
```
@model DayOfWeek

@switch(Model)
{
    case DayOfWeek.Monday:
        <text>Maandag</text>
        break;
    case DayOfWeek.Tuesday:
        <text>Dinsdag</text>
        break;
    case DayOfWeek.Wednesday:
        <text>Woensdag</text>
        break;
    case DayOfWeek.Thursday:
        <text>Donderdag</text>
        break;
    case DayOfWeek.Friday:
        <text>Vrijdag</text>
        break;
    case DayOfWeek.Saturday:
        <text>Zaterdag</text>
        break;
    case DayOfWeek.Sunday:
        <text>Zondag</text>
        break;
}
```

This was perfectly fine in Mvc 1.0.1, but in 1.1 it results in:

`InvalidOperationException: The model item passed into the ViewDataDictionary is of type 'System.String', but this ViewDataDictionary instance requires a model item of type 'System.DayOfWeek'.`

Maybe this is By Design with the changes in enum and DisplayName and I missed this as a breaking change. Still I think it's strange that an Enum becomes a string in the ViewDataDictionary.

---
Guess guess => maybe this change has something to do with it: 830983a#diff-a3619a957ef987ec3d82c9fe445320e9

**Comment by tverboon** [comment]
Guess guess => maybe this change has something to do with it: https://github.com/aspnet/Mvc/commit/830983a477e19cfb7527e1caba3624a8424397a6

**Comment by Eilon** [comment]
@ryanbrandenburg can you investigate?

**Comment by drauch** [comment]
@ryanbrandenburg : When is this going to be released? It seems to be a problem still (using AspNetCore.Mvc 1.1.3)

**Comment by Eilon** [comment]
@drauch - I believe this fix is in the 2.0.0 release, which will be available shortly. We have nightly builds that you can use to test.

**Comment by tverboon** [comment]
Yes, it's fixed in 2.0.0-preview.

**Comment by fschmied** [comment]
@Eilon Any chance of getting the fix in an 1.1 patch release?

**Comment by Eilon** [comment]
@fschmied unfortunately at this time we are not planning to back-port this to 1.1.x. Perhaps you could use a workaround such as using another data type to represent the enum in the ViewData, such as an integer? (E.g. case the enum to an int, then use the int in the CSHTML.)
